cypress/e2e/build-your-plate.cy.ts

```diff
@@ -1,6 +1,257 @@
-  it("gives correct score with 5 meats");
-  it("gives correct score with 5 mixed items");
-  it("allows user to swap food items before scoring");
+  beforeEach(() => {
+    cy.visit("http://localhost:3000/Games/BuildYourPlate");
+  });
+
+  it("check a11y buuld-your-plate page", () => {
+    cy.injectAxe();
+    cy.checkA11y(null, null, terminalLog, true);
+  });
+
+  it("can navigate to the game page", () => {
+    cy.visit("http://localhost:3000/");
+    cy.get(".react-multi-carousel-list [data-testid='Games']").click();
+    cy.get("[data-testid='menu-styling'] [data-testid='Build your plate']").click();
+    cy.url().should("eq", "http://localhost:3000/Games/BuildYourPlate");
+  });
+
+  it("gives correct score with 5 sweets", () => {
+    cy.get("[data-testid='game-begin']").click();
+    cy.contains("button", "Play").click();
+    cy.contains("button", "Score my plate").should("be.disabled");
+    cy.contains("0 / 5");
+    cy.get("button [alt='Sweets']", { timeout: 10000 }).click();
+
+    Object.keys(foodList.sweets).forEach((item) => {
+      cy.get("body").then((body) => {
+        if (body.find(`button [alt='${item}']`).length) {
+          cy.get(`button [alt='${item}']`).click();
+        }
+      });
+    });
+
+    cy.get("[alt='Tick']").should("have.length", 5);
+    cy.contains("button", "Score my plate").should("be.enabled");
+    cy.contains("5 / 5");
+    cy.contains("button", "Score my plate").click();
+    cy.get("[alt='plate']");
+    cy.contains("button", "Score my plate").click();
+    cy.contains("Score: 10 out of 50");
+    cy.contains("Could be better!");
+    cy.contains("button", "Play again").click();
+  });
+
+  it("gives correct score with 5 meats", () => {
+    cy.get("[data-testid='game-begin']").click();
+    cy.contains("button", "Play").click();
+    cy.contains("button", "Score my plate").should("be.disabled");
+    cy.contains("0 / 5");
+    cy.get("button [alt='Meat']", { timeout: 10000 }).click();
+
+    const expectedScore = [0];
+    cy.wrap(expectedScore).as("expectedScore");
+    Object.entries(foodList.meats).forEach(([key, value]) => {
+      cy.get("body").then((body) => {
+        if (body.find(`button [alt='${key}']`).length) {
+          cy.get(`button [alt='${key}']`).click();
+          cy.get<number[]>("@expectedScore").then((score) => {
+            score.push(score.pop() + value);
+          });
+        }
+      });
+    });
+
+    cy.get("[alt='Tick']").should("have.length", 5);
+    cy.contains("button", "Score my plate").should("be.enabled");
+    cy.contains("5 / 5");
+    cy.contains("button", "Score my plate").click();
+    cy.get("[alt='plate']");
+    cy.contains("button", "Score my plate").click();
+    cy.get<number[]>("@expectedScore").then((score) => {
+      cy.contains(`Score: ${score[0]} out of 50`);
+    });
+    cy.contains("Great!");
+    cy.contains("button", "Play again").click();
+  });
+
+  it("gives correct score with 5 mixed items", () => {
```

Comment:
Unsure about having a test with a randomised output

Reply:
Note - discussed this briefly with Andrew and we've agreed that the current approach is our best option at present. While there may be alternative options for handling the randomness, we don't have one that would be better, either requiring substantial reworking beyond the scope of this ticket or requiring methods previously discussed as impractical.